package.json

```diff
@@ -13,8 +13,7 @@
   },
   "files": [
     "haunted.js",
-    "index.js",
-    "web.js",
+    "src",
```

Comment:
So since this in JavaScript there is no "build". So it would just be a direct copy. That seems overly complicated. If your issue is the src/ name would it be better if this was renamed to lib/ ??

Reply:
yep, it would just be a copy step

i see a build involving a makefile, so i'm thinking of a copy step

regarding `lib/`, there was some discussion about that in #98 

> > Good point on src/, could possibly rename to lib/.
> 
> some people like sourcecode in src, and distribution code in dist
> some people prefer to use the term lib instead of dist
> the problem, is that some people like to use lib instead of src, and then emit to dist
> so when we see somebody use lib, we can't be sure if that's sourcecode, or if that's consumable output
> if lib is where the sourcecode lives -- then of course we should never point to lib, because we want to be able to refactor the sourcecode safely without affecting the emitted output
> 
> if you want to call the consumable output lib, then there's this minor ambiguity problem
> 
> so it seems like the hip thing to do is to keep it simple and unambiguous -- src for sourcecode, dist for what gets distributed -- and we just never use lib, in order to avoid the confusion altogether

so yes, a copy step to `lib/` would be good, but i think a copy step to `dist/` or just dumped in the project root might be better

if we dump things in the project root, the benefit is nice import paths, eg, `"haunted/core.js"` instead of `"haunted/dist/core.js"`

all that really matters is that it's not in a place called `src`, which should be reserved as off-limits to consumers, such that any potential future refactor is possible

Reply:
`haunted/core` is supported here because haunted/core.js exists.  I'm not particularly worried about people poking inside of `lib/` or `src/` or whatever. The same problem would exist for `dist/`, people could poke inside and then you're not allowed to "break" that API.  So simply copying a folder doesn't avoid that problem.  I think it's more simple to say that the exposed places to import are:

* `haunted` which resolves from package.json
* `haunted/core` which resolves to `haunted/core.js`
* `haunted/haunted.js` the ESM bundle.

If we say those are the only supported imports, then someone doing `haunted/some/folder/whatever.js` is not supported.

Reply:
> `haunted/core` is supported here because `haunted/core.js` exists.

ah yes, of course, i see now — that means moving to a `dist/` or `lib/` directory would be a breaking change, and we don't want that at all

the project root would continue to be the right place to expose code :+1:

> I'm not particularly worried about people poking inside of `lib/` or` src/` or whatever. The same problem would exist for `dist/`, people could poke inside and then you're not allowed to "break" that API. So simply copying a folder doesn't avoid that problem.
>
> If we say those are the only supported imports, then someone doing `haunted/some/folder/whatever.js` is not supported.

ahah, i think i get what you're saying: that if we use alias files like `haunted/core.js` re-exporting stuff from `src/`, then for example, a typescript conversion could still go smoothly, because the alias files could simply start pointing to `dist/` instead, which should be opaque to the user — these root-level re-export aliases demarcate the public signature

okay that makes sense, i get it now :+1:

> - `haunted/core` which resolves to `haunted/core.js`

i think you mean `haunted/core.js` re-exports `haunted/src/core.js`

> - `haunted` which resolves from package.json

hmm.. looking at the package.json main/module right now, i'm seeing `index.js`..

wait, didn't you delete `index.js`? is this a mistake?

it looks to me like the ideal main/module is `haunted/src/haunted.js`

Reply:
Right, `haunted/src/haunted.js` should be the new main. I do still like the idea of renaming `src/` to `lib/` anyways. But yes, the `lib/` folder is not something the user will ever reach into, if they want core they will get `haunted/core.js` and that is the demarcation for the public API.

Will update the PR with moving to `lib/` and fix the package.json main/module.  And bring back in `index.js` but mark it as deprecated, to keep this as a minor version change.